### PR TITLE
Add logging in pipeline to identify error

### PR DESCRIPTION
**Pull request by ajaygargdev**
Added Log to identify errors on deploy infrastructure.

Previously we were not getting the error in deploying infrastructure and the task was also not failing like failing to create any resource.

image

The tested scenario is :
       Create Postgres Database without a created database server. Previously we are not getting any error but now we are able to get errors. Like:-

image

### CODE REVIEW

File: deploy/deploy-infrastructure.ps1 [hunk 1]
```diff
@@ -94,8 +94,11 @@ $ResourceNamePrefix = "kmd-momentum-mea-$InstanceId"
 $TemplateFile = "azuredeploy.json"
 
 try {
-  [Microsoft.Azure.Common.Authentication.AzureSession]::ClientFactory.AddUserAgent("VSAzureTools-$UI$($host.name)".replace(' ','_'), '3.0.0')
-} catch { }
+  #[Microsoft.Azure.Common.Authentication.AzureSession]::ClientFactory.AddUserAgent("VSAzureTools-$UI$($host.name)".replace(' ','_'), '3.0.0')
+} catch { 
+  Write-Host "An error occurred:"
+  Write-Host $_
```
Comment (ExploreMyCode) [inline deploy/deploy-infrastructure.ps1:100]: Can you please try to capture error as well ? something like 
catch(Exceptino ex)
{
write-host "An error occurred : " + ex.tostring();
}

File: deploy/deploy-infrastructure.ps1 [hunk 2]
```diff
@@ -135,34 +138,44 @@ $TemplateParameters = @{
...
 
-Pop-Location
\ No newline at end of file
+		if ($ErrorMessages) {
+			Write-Output '', 'Template deployment returned the following errors:', @(@($ErrorMessages) | ForEach-Object { $_.Exception.Message.TrimEnd("`r`n") })
+			exit 1
+		}
+	}
+}catch{
+	Write-Host "An error occurred:"
```
Comment (ExploreMyCode) [inline deploy/deploy-infrastructure.ps1:176]: Same as above.

File: deploy/deploy-infrastructure.ps1 (hunk 1 above)
Comment (ajaygargdev) [reply deploy/deploy-infrastructure.ps1:100]: $_ contains the exception so we are using $_ for exception

File: deploy/deploy-infrastructure.ps1 (hunk 2 above)
Comment (ajaygargdev) [reply deploy/deploy-infrastructure.ps1:176]: $_ contains Exception, log so we are writing log using $_

File: deploy/deploy-infrastructure.ps1 [hunk 3]
```diff
@@ -135,34 +138,44 @@ $TemplateParameters = @{
 # Create or update the resource group using the specified template file and template parameter values
 $Tags = @{}
 if ($MarkForAutoDelete) {
-  $Tags["keep"] = "false";
+	$Tags["keep"] = "false";
```
Comment (amitxagarwal) [inline deploy/deploy-infrastructure.ps1:141]: why these unnecessary changes? Looks like indentation got corrected here. If that is the case then why this was not taken care when created this script in the initial phase?

**Comment by amitxagarwal** [comment]
With every PR like this, we need to clearly mention what kind of error scenarios we have tested these scripts with. If no scenario tested then we need a good reason for the same.

**Comment by amitxagarwal** [comment]
Also once these changes are tested against different erroneous scenarios it is always good to mention the screenshots for the same so that it becomes easy for others to understand.

File: deploy/deploy-infrastructure.ps1 (hunk 3 above)
Comment (ajaygargdev) [reply deploy/deploy-infrastructure.ps1:141]: did some changes related to format to understand the script properly
